bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/GeyserModBootstrap.java

```diff
@@ -242,7 +245,19 @@ public String getServerBindAddress() {
 
     @Override
     public int getServerPort() {
-        return ((GeyserServerPortGetter) server).geyser$getServerPort();
+        if (isServer()) {
+            return ((GeyserServerPortGetter) server).geyser$getServerPort();
+        } else {
+            // Set in the IntegratedServerMixin
+            return geyserConfig.getRemote().port();
+        }
+    }
+
+    public abstract boolean isServer();
```

Comment:
For this, can we use `Platform.getEnvironment()`? https://docs.architectury.dev/api/platform

Reply:
This would require us to ask users to install the architectury API mod - for just one method, that seems overkill

Reply:
I see, my bad. I thought it transpiled code to the right version on compile.

Reply:
That would be archloom's `@ExpectPlatform` annotation:
https://docs.architectury.dev/plugin/expect_platform
However, this would still require adding it on the fabric/NeoForge module.. so imo it's easier to just use an abstract method + impls